src/main/scripts/src/app/shared/search-box/search-box.component.ts

```diff
@@ -32,44 +35,55 @@ import 'rxjs/add/operator/map';
 import 'rxjs/add/operator/mergeMap';
 
 import {Suggestions} from '../../core/dto/suggestions';
-import {SearchService} from '../../core/rest/search.service';
 import {WorkspaceService} from '../../core/workspace.service';
 import {QueryItem} from './query-item/query-item';
 import {FulltextQueryItem} from './query-item/fulltext-query-item';
 import {AttributeQueryItem} from './query-item/attribute-query-item';
 import {CollectionQueryItem} from './query-item/collection-query-item';
-import {SuggestionType} from '../../core/dto/suggestion-type';
 import {Collection} from '../../core/dto/collection';
 import {QueryItemsConverter} from './query-item/query-items-converter';
 import {Query} from '../../core/dto/query';
 import {QueryConverter} from '../utils/query-converter';
 import {ViewService} from '../../core/rest/view.service';
 import {KeyCode} from '../key-code';
 import {HtmlModifier} from '../utils/html-modifier';
+import {SearchMockService} from '../../core/rest/search-mock.service';
+import {QueryItemType} from './query-item/query-item-type';
+import {ConditionQueryItem} from './query-item/condition-query-item';
 
 @Component({
   selector: 'search-box',
   templateUrl: './search-box.component.html',
   styleUrls: ['./search-box.component.scss'],
   providers: [QueryItemsConverter]
 })
-export class SearchBoxComponent implements OnInit {
+export class SearchBoxComponent implements OnInit, AfterViewInit {
+
+  @ViewChildren('queryItemCondition')
+  public conditions: QueryList<ElementRef>;
+
+  @ViewChild('searchBox')
+  public input: ElementRef;
 
   public queryItems: QueryItem[] = [];
-  public currentQueryItem: QueryItem = new FulltextQueryItem('');
```

Comment:
The idea behind this was that everything added to `queryItems` must be already completed query item. Otherwise, it stays in the editing mode and is just stored in `currentQueryItem`. Why have you changed this?

Reply:
My idea is to put everything to `quertItems` and show some kind of warning if it's not completed.